### PR TITLE
EL9 - Retrieve python3-pyOpenSSL from epel

### PR DESCRIPTION
**Pull request by austlane**
##### SUMMARY

With the release of CentOS 9 / RHEL 9. `python3-pyOpenSSL` has been moved from the `AppStream` repo to `epel`.

Currently, attempting to include `community.crypto` in an el9-based Execution Environment results in failure. This PR corrects the issue using the `epel` ansible-builder profile documented here https://github.com/ansible/ansible-builder/issues/156

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

bindep dependency resolution

### CODE REVIEW

**Comment by felixfontein** [comment]
Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

I'll take a closer look at the PR later.

File: meta/ee-bindep.txt [hunk 1]
```diff
@@ -11,4 +11,5 @@ openssl [platform:rpm]
 python3-cryptography [platform:dpkg]
 python3-cryptography [platform:rpm]
 python3-openssl [platform:dpkg]
-python3-pyOpenSSL [platform:rpm]
+python3-pyOpenSSL [platform:rpm !platform:rhel-9 !platform:centos-9]
+python3-pyOpenSSL [platform:rhel-9 platform:centos-9 epel]
```
Comment (felixfontein) [inline meta/ee-bindep.txt:15]: How about the following:
```suggestion
# On RHEL 9+ and CentOS Stream 9+, python3-pyOpenSSL is part of EPEL
python3-pyOpenSSL [platform:rpm !platform:rhel !platform:centos]
python3-pyOpenSSL [platform:rhel-6 platform:centos-6 platform:rhel-7 platform:centos-7 platform:rhel-8 platform:centos-8]
python3-pyOpenSSL [platform:rhel platform:centos !platform:rhel-6 !platform:centos-6 !platform:rhel-7 !platform:centos-7 !platform:rhel-8 !platform:centos-8 epel]
```
1. For non-RHEL/CentOS RPM-based systems (like OpenSuSE), `python3-pyOpenSSL` from regular packages will be used;
2. For RHEL/CentOS 6, 7, and 8, `python3-pyOpenSSL` from regular packages will be used;
3. For RHEL/CentOS other than 6, 7, and 8, `python3-pyOpenSSL` from `epel` will be used.

**Comment by austlane** [comment]
Incorporated suggested changes from @felixfontein and added a changelog fragment. Should be good to go 👍

File: changelogs/fragments/575-bindep-python3-pyOpenSSL.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - "bindep - Fix `python3-pyOpenSSL` dependency resolution on el9 platforms (https://github.com/ansible-collections/community.crypto/pull/575)."
```
Comment (felixfontein) [inline changelogs/fragments/575-bindep-python3-pyOpenSSL.yml:2]:
```suggestion
  - "execution environment binary dependencies (bindep.txt) - fix ``python3-pyOpenSSL`` dependency resolution on RHEL 9+ / CentOS Stream 9+ platforms (https://github.com/ansible-collections/community.crypto/pull/575)."
```

**Comment by felixfontein** [comment]
Did you test the changes on RHEL 9 and/or CentOS Stream 9? The EE base images used in CI are all CentOS Stream 8 based, so they only test part of this.

**Comment by felixfontein** [comment]
Oh, and could you please rebase and remove the `@` from the commit message? Having `@felixfontein` in there will cause me to get notifications every single time someone pushes this commit to their repo (like when updating their fork of this repo to the latest state once this PR is merged), and I'd prefer not to get that stream of notifications :D

**Comment by felixfontein** [comment]
@austlane ping - if you want this to appear in the next release please answer soon. I want to do a release today or tomorrow.

**Comment by austlane** [comment]
@felixfontein rebased with the `@` removed!

I have tested this against centos 9, which is used in `awx-ee`
https://github.com/ansible/awx-ee

**Comment by felixfontein** [comment]
> I have not tested against RHEL9 (as used in the Redhat Ansible Automation Platform), as I do not have a subscription cry

Same here ;) I would expect it to behave the same...

**Comment by felixfontein** [comment]
@austlane thanks a lot for fixing this!

**Pull request merged**